src/mon/FSCommands.cc

```diff
@@ -904,6 +904,16 @@ class AddDataPoolHandler : public FileSystemCommandHandler
       }
     }
 
+    for (auto& fs : fsmap.get_filesystems()) {
+      // not allowed if poolid is already in use by any fs
+      const std::vector<int64_t> &data_pools = fs->mds_map.get_data_pools();
+      if (std::find(data_pools.begin(), data_pools.end(), poolid) != data_pools.end()
```

Comment:
you could also change this to:

    auto it = std::find(data_pools.begin(), data_pools.end(), poolid);
    if (it != data_pools.end() || fs->mds_map.get_metadata_pool() == poolid) {
      ...
    }

Reply:
Done